services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/skills_report/performance_breakdown/by_student.tsx

```diff
@@ -0,0 +1,460 @@
+}
+
+interface SmartGroupButtonProps {
+  smartGroupName: string;
+  count: number;
+  colorClass: 'green' | 'gold' | 'grey';
+  onQuickAssign: (smartGroupName: string) => void;
+}
+
+const SmartGroupButton: React.FC<SmartGroupButtonProps> = ({ smartGroupName, count, colorClass, onQuickAssign }) => {
+  const handleClick = React.useCallback(() => onQuickAssign(smartGroupName), [onQuickAssign, smartGroupName]);
+
+  return (
+    <button 
+      className={`quill-button small outlined ${colorClass} focus-on-light`}
+      onClick={handleClick} 
+      type="button"
+    >
+      {smartGroupName} ({count})
+    </button>
+  );
+};
+
+const ByStudent: React.FC<ByStudentProps> = ({ 
+  classroom, 
+  momentumScoreGoal, 
+  practiceScope, 
+  selectedSkillCategory, 
+  selectedSkillSubCategory 
+}) => {
+  const [data, setData] = React.useState<{[concept_id: number]: StudentPerformanceData[]}>({})
+  const [selectedRows, setSelectedRows] = React.useState<(string | number)[]>([])
+  const [dataKey, setDataKey] = React.useState<number>(selectedSkillSubCategory?.id || selectedSkillCategory?.id)
+  
+  const tableRef = React.useRef<HTMLTableElement>(null)
+  const wrapperRef = React.useRef<HTMLDivElement>(null)
+  const containerRef = React.useRef<HTMLDivElement>(null)
+
+  React.useEffect(() => {
+    const newDataKey = selectedSkillSubCategory?.id || selectedSkillCategory?.id
+    setDataKey(newDataKey)
+
+    const queryHash = {
+      concept_id: selectedSkillCategory?.id,
+      sub_concept_id: selectedSkillSubCategory?.id,
+      classroom_id: classroom.id,
+      all_history: practiceScope === ALL_PRACTICE
+    };
+
+    const url = queryString.stringifyUrl(
+      {
+        query: queryHash,
+        url: '/skills_reports/performance_breakdown_by_student',
+      },
+      { arrayFormat: 'bracket' }
+    );
+    
+    requestGet(url, (body) => {
+      setSelectedRows([])
+      setData({...data, [newDataKey]: body.data})
+    })
+  }, [classroom, selectedSkillCategory, selectedSkillSubCategory, practiceScope])
+
+  React.useEffect(() => {
+    setData({})
+  }, [classroom, practiceScope])
+
+  const columnWidths = [
+    { width: 300, minWidth: 170 }, // Student Name
+    { width: 200, minWidth: 140 }, // Smart Group
+    { width: 200, minWidth: 110 }, // Momentum Score
+    { width: 300, minWidth: 140 }, // Momentum Score Trend
+    { width: 200, minWidth: 116 }, // Exposure
+    { width: 200, minWidth: 116 }, // Questions Answered
+    { width: 200, minWidth: 116 }, // Replay Rate
+    { width: 200, minWidth: 116 }, // Time Spent
+    { width: 200, minWidth: 100 }, // Last Practiced
+  ];
+
+  const scaledWidths = useScaledTableColumnWidths({
+    columnWidths,
+    containerRef,
+    maxTableWidth: 2000,
+    minTableWidth: 1200,
+    additionalWidthRestrictions: 51
+  });
+
+  useStickyTableHeader({
+    wrapperRef,
+    tableRef,
+    enabled: true,
+    dependencies: [data],
+  });
+
+  const formatTime = (seconds: number) => {
```

Comment:
how about renaming it to `secondsToStringHoursAndMinutes`? It's more easily searchable for future features that may want to recycle this logic